File: src/test/java/org/thoughtworks/linesh/multicurrencymoney/DollarTest.java
package org.thoughtworks.linesh.multicurrencymoney;

import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;
import support.Tasking;

import static org.junit.jupiter.api.Assertions.*;

class DollarTest {
    @Tasking @Nested
    class should_test_multiplications {
        @Test
        void should_test_dollar_multiplications() {
            Money five = Money.dollar(5);

            // It's actually a risk here that we rely on equality function for tests to work
            assertEquals(Money.dollar(10), five.times(2));
            assertEquals(Money.dollar(15), five.times(3));
        }
        
        @Tasking @Nested
        class should_test_equality {
            @Test
            void should_test_dollar_equality() {
                assertTrue(Money.dollar(5).equals(Money.dollar(5)));
                assertFalse(Money.dollar(5).equals(Money.dollar(6)));
                assertFalse(Money.dollar(5).equals(null));
            }
        }
    }

    @Tasking
    @Nested
    class should_add_two_different_kinds_of_currency_given_exchange_rate {
        
    }

}

File: test/multi-currency/Dollar.test.ts
import * as assert from 'assert'
import Dollar from '../../src/multi-currency/Dollar'
import Money from '../../src/multi-currency/Money'

it('multiplication', () => {
  assert(Money.dollar(5).times(2).equals(Money.dollar(10)))
  assert(Money.dollar(5).times(3).equals(Money.dollar(15)))
})

it('money equality', () => {
  assert(new Dollar(5).equals(new Dollar(5)))
  assert.equal(new Dollar(5).equals(new Dollar(4)), false)
  assert.equal(new Dollar(5).equals(null), false)
})
